dspace-api/src/main/java/org/dspace/authenticate/ShibAuthentication.java

```diff
@@ -226,7 +226,12 @@ public int authenticate(Context context, String username, String password,
 			// Step 4: Log the user in.
 			context.setCurrentUser(eperson);
 			request.getSession().setAttribute("shib.authenticated", true);
-            authenticationService.initEPerson(context, request, eperson);
+
+			//When testing, this object did not seem to be properly initialized on the first test
+			if (authenticationService == null) {
+				authenticationService = AuthenticateServiceFactory.getInstance().getAuthenticationService();
```

Comment:
This variable was not properly initialized when this code was invoked.  Is there a cleaner way to ensure that the proper objects have been initialized?